client.go

```diff
@@ -675,10 +690,10 @@ func (options *ClientOptions) withDefaults() (updated *ClientOptions) {
 	*updated = *options
 	// impose defaults
 	if updated.SocketTimeout <= 0 {
-		updated.SocketTimeout = time.Second * 300
+		updated.SocketTimeout = time.Second * 10000
```

Comment:
Isn't this timeout limit too high?

Reply:
oops, sorry, that snuck in from some changes I made to support another project